### PR TITLE
[image] Add SVG support

### PR DESCRIPTION
**Pull request by jgonet**
# Why

To extend supported formats.
Superseding #7385.

# How

Used SDWebImageSVGKitPlugin & androidsvg, copied SVG example from Glide repository.
Currently, there's a problem with SVGKit which doesn't compile on the 3.0 version (required by a newer version of SVGKitPlugin) so I didn't allow changing plugin version on iOS.

# Test Plan

Tested on bare-expo. Note I didn't update Podfile.lock - bare-expo doesn't work well with Flipper and I applied a patch removing it which changes the state of the lock file.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDecoder.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  @Throws(IOException::class)
+  override fun decode(source: InputStream, width: Int, height: Int, options: Options): Resource<SVG>? {
+    return try {
+      SimpleResource(SVG.getFromInputStream(source))
+    } catch (ex: SVGParseException) {
+      throw IOException("Cannot load SVG from stream", ex)
+    }
+  }
+}
```
Comment (bbarthec) [inline packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDecoder.kt:31]:
```suggestion
}

```

File: packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDrawableTranscoder.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import com.bumptech.glide.load.resource.SimpleResource
+import com.bumptech.glide.load.resource.transcode.ResourceTranscoder
+import com.caverock.androidsvg.SVG
+
+/**
+ * Convert the [SVG]'s internal representation to an Android-compatible one ([Picture]).
+ *
+ *
+ * Copied from https://github.com/bumptech/glide/blob/10acc31a16b4c1b5684f69e8de3117371dfa77a8/samples/svg/src/main/java/com/bumptech/glide/samples/svg/SvgDrawableTranscoder.java
+ * and rewritten to Kotlin
```
Comment (bbarthec) [inline packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDrawableTranscoder.kt:16]:
```suggestion
 * Convert the [SVG]'s internal representation to an Android-compatible one ([Picture]).
 *
 * Copied from https://github.com/bumptech/glide/blob/10acc31a16b4c1b5684f69e8de3117371dfa77a8/samples/svg/src/main/java/com/bumptech/glide/samples/svg/SvgDrawableTranscoder.java
 * and rewritten to Kotlin.
```

File: packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDrawableTranscoder.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+class SVGDrawableTranscoder : ResourceTranscoder<SVG?, Drawable> {
+  override fun transcode(toTranscode: Resource<SVG?>, options: Options): Resource<Drawable> {
+    val picture = toTranscode.get().renderToPicture()
+    val drawable: Drawable = SVGDrawable(picture)
+    return SimpleResource(drawable)
+  }
+}
```
Comment (bbarthec) [inline packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDrawableTranscoder.kt:24]:
```suggestion
}

```

File: packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGModule.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.bumptech.glide.module.LibraryGlideModule
+import com.caverock.androidsvg.SVG
+import java.io.InputStream
+
+/**
+ * [LibraryGlideModule] registering support for SVG to Glide.
+ *
+ *
+ * Copied from https://github.com/bumptech/glide/blob/10acc31a16b4c1b5684f69e8de3117371dfa77a8/samples/svg/src/main/java/com/bumptech/glide/samples/svg/SvgModule.java
+ * and rewritten to Kotlin
```
Comment (bbarthec) [inline packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGModule.kt:17]:
```suggestion
 * [LibraryGlideModule] registering support for SVG to Glide.
 *
 * Copied from https://github.com/bumptech/glide/blob/10acc31a16b4c1b5684f69e8de3117371dfa77a8/samples/svg/src/main/java/com/bumptech/glide/samples/svg/SvgModule.java
 * and rewritten to Kotlin.
```

File: packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGModule.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ */
+@GlideModule
+class SVGModule : LibraryGlideModule() {
+  override fun registerComponents(context: Context, glide: Glide, registry: Registry) {
+    super.registerComponents(context, glide, registry)
+    registry
+      .append(InputStream::class.java, SVG::class.java, SVGDecoder())
+      .register(SVG::class.java, Drawable::class.java, SVGDrawableTranscoder())
+  }
+}
```
Comment (bbarthec) [inline packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGModule.kt:27]:
```suggestion
}

```

File: packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGSoftwareLayerSetter.kt [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import com.bumptech.glide.request.target.ImageViewTarget
+import com.bumptech.glide.request.target.Target
+
+/**
+ * Listener which updates the [ImageView] to be software rendered, because [ ]/[android.graphics.Picture] can't render on a
+ * hardware backed [Canvas][android.graphics.Canvas].
+ *
+ *
+ * Copied from https://github.com/bumptech/glide/blob/10acc31a16b4c1b5684f69e8de3117371dfa77a8/samples/svg/src/main/java/com/bumptech/glide/samples/svg/SvgSoftwareLayerSetter.java
+ * and rewritten to Kotlin
```
Comment (bbarthec) [inline packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGSoftwareLayerSetter.kt:16]:
```suggestion
 * Copied from https://github.com/bumptech/glide/blob/10acc31a16b4c1b5684f69e8de3117371dfa77a8/samples/svg/src/main/java/com/bumptech/glide/samples/svg/SvgSoftwareLayerSetter.java
 * and rewritten to Kotlin.
```

File: packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGSoftwareLayerSetter.kt [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  private fun getViewOfTarget(target: Target<Any>): ImageView? {
+    if (target is ImageViewTarget<*>) {
+      val imageViewTarget: ImageViewTarget<*> = target as ImageViewTarget<Any>
+      return imageViewTarget.view
+    }
+    return null
+  }
+}
```
Comment (bbarthec) [inline packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGSoftwareLayerSetter.kt:40]:
```suggestion
}

```

**Review by bbarthec — APPROVED** [review]
Looks fine to me 👍 
I'll try it myself and see how it works and then I'll merge 💪

**Review by bbarthec — CHANGES_REQUESTED** [review]
I believe I've spotted a problem 🤔
I've tried running the example screen on both platforms and obtained different results. 

On iOS image is not sharp 🤔 

Android | iOS
-|-
 |

**Comment by jakub-gonet** [comment]
Sorry for the pings, pull with rebase messed up history a bit 😞

**Pull request merged**